operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -200,9 +183,30 @@ type CassandraNodeStatus struct {
 
 type CassandraStatusMap map[string]CassandraNodeStatus
 
+type DatacenterConditionType string
+
+const (
+	DatacenterReady          DatacenterConditionType = "Ready"
+	DatacenterInitialized    DatacenterConditionType = "Initialized"
+	DatacenterReplacingNodes DatacenterConditionType = "ReplacingNodes"
+	DatacenterScalingUp      DatacenterConditionType = "ScalingUp"
+	DatacenterUpdating       DatacenterConditionType = "Updating"
+	DatacenterStopped        DatacenterConditionType = "Stopped"
+	DatacenterResuming       DatacenterConditionType = "Resuming"
+	DatacenterRollingRestart DatacenterConditionType = "RollingRestart"
+)
+
+type DatacenterCondition struct {
+	Type DatacenterConditionType `json:"type"`
+	Status corev1.ConditionStatus `json:"status"`
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
```

Comment:
did you ever figure out what was wrong with updating the slice? if you had a zero value here, that could be part of the issue, like here https://github.com/coreos/prometheus-operator/issues/2399

Reply:
I never did figure out what was going on with updating the slice, I actually need to check to see if it is still necessary to recreate the array. That timestamp was added after dealing with those issues though. I haven't seen this cause a problem yet, possibly because it never ends up being null?

Reply:
Just tested this. It works fine without creating a new array.